### PR TITLE
[top] Update cpu_info test

### PR DESCRIPTION
**Pull request by tjaychen**
Due to #15219, it appears the current PC information on double faults is now changed. Update to reflect the latest.

Also change illegal read to an illegal write at address 0, since this reduces the number of "glue" instructions surrounding the one that actually creates the illegal access.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
hey all can you help have a look?
I feel like this test behaves slightly differently now because we pulled in the ibex fetch_en fix very recently. 
let me know if what i did here makes sense.

File: sw/device/tests/rstmgr_cpu_info_test.c [hunk 1]
```diff
@@ -272,23 +273,20 @@ bool test_main(void) {
...
             "CPU Info dump doesn't show a double fault has happened.");
 
-      // The current behaviour after a double fault is to capture, in the CPU
-      // info dump, the interrupt vector below the one which was taken to jump
-      // to the exception handler as the current PC and the start of the
-      // exception handler as the next PC. This feels wrong. However, with a
-      // lack of a clear definition of what these values should contain, the
-      // test enforces this behaviour so that regressions can be caught.
-      uint32_t curr_pc = (uint32_t)&_ottf_interrupt_vector + 4;
-      uint32_t next_pc = (uint32_t)&handler_exception;
```
Comment (HU90m) [inline sw/device/tests/rstmgr_cpu_info_test.c:282]: The `handle_exception` symbol was only used here, so can be removed from the rest of the file.

File: sw/device/tests/rstmgr_cpu_info_test.c [hunk 2]
```diff
@@ -79,8 +79,9 @@ volatile static bool double_fault;
 void ottf_exception_handler(void) {
   if (double_fault) {
     OT_ADDRESSABLE_LABEL(kDoubleFaultSecondAddrLower);
-    addr_val = mmio_region_read32(mmio_region_from_addr(kIllegalAddr2), 0);
+    mmio_region_write32(mmio_region_from_addr(kIllegalAddr2), 0, 0);
     OT_ADDRESSABLE_LABEL(kDoubleFaultSecondAddrUpper);
+    mmio_region_write32(mmio_region_from_addr(kIllegalAddr2), 0, 0);
```
Comment (HU90m) [inline sw/device/tests/rstmgr_cpu_info_test.c:84]: I couldn't work out what this second write is for. It is to make sure the address of the next instruction field of the cpu info dump is known?

Comment (tjaychen) [reply sw/device/tests/rstmgr_cpu_info_test.c:84]: o wait i dont need that anymore, let me remove it.
This was back when I thought the next_pc would still be the next instruction when it exceptioned.

File: sw/device/tests/rstmgr_cpu_info_test.c (hunk 1 above)
Comment (tjaychen) [reply sw/device/tests/rstmgr_cpu_info_test.c:282]: removed!

**Pull request merged**